packages/core/index.test-d.ts

```diff
@@ -196,3 +205,178 @@ expectType<middy.MiddyfiedHandler<unknown>>(streamifiedResponseHandler)
+  beforePrefetch () { console.log('beforePrefetch') },
+  requestStart () { console.log('requestStart') },
+  beforeMiddleware (name: string) { console.log('beforeMiddleware', name) },
+  afterMiddleware (name: string) { console.log('afterMiddleware', name) },
+  beforeHandler () { console.log('beforeHandler') },
+  afterHandler () { console.log('afterHandler') },
+  async requestEnd () { console.log('requestEnd') }
+})
+expectType<Handler>(syncedHandler)
+
+// invokes the handler to test that it is callable
+// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+async function invokSyncedHandler (): Promise<void | APIGatewayProxyResult> {
+  const sampleEvent: APIGatewayProxyEvent = {
+    resource: '/',
+    path: '/',
+    httpMethod: 'GET',
+    requestContext: {
+      resourcePath: '/',
+      httpMethod: 'GET',
+      path: '/Prod/',
+      accountId: 'x',
+      apiId: 'y',
+      authorizer: {},
+      protocol: 'p',
+      identity: {
+        accessKey: '',
+        accountId: '',
+        apiKey: '',
+        apiKeyId: '',
+        caller: '',
+        clientCert: null,
+        cognitoAuthenticationProvider: '',
+        cognitoAuthenticationType: '',
+        cognitoIdentityId: '',
+        cognitoIdentityPoolId: '',
+        principalOrgId: '',
+        sourceIp: '',
+        user: '',
+        userAgent: '',
+        userArn: ''
+      },
+      stage: '',
+      requestId: '',
+      requestTimeEpoch: 12345567,
+      resourceId: ''
+    },
+    headers: {
+      accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+      'accept-encoding': 'gzip, deflate, br',
+      Host: '70ixmpl4fl.execute-api.us-east-2.amazonaws.com',
+      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36',
+      'X-Amzn-Trace-Id': 'Root=1-5e66d96f-7491f09xmpl79d18acf3d050'
+    },
+    multiValueHeaders: {
+      accept: [
+        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9'
+      ],
+      'accept-encoding': [
+        'gzip, deflate, br'
+      ]
+    },
+    queryStringParameters: null,
+    multiValueQueryStringParameters: null,
+    pathParameters: null,
+    stageVariables: null,
+    body: null,
+    isBase64Encoded: false
+  }
+  const sampleContext: Context = {
+    callbackWaitsForEmptyEventLoop: true,
+    functionName: '',
+    functionVersion: '',
+    invokedFunctionArn: '',
+    memoryLimitInMB: '234',
+    awsRequestId: '',
+    logGroupName: '',
+    logStreamName: '',
+    getRemainingTimeInMillis: (): number => 1,
+    done: () => { },
+    fail: (_) => { },
+    succeed: () => { }
+  }
+  // synced handler but could be void | Promise<TResult> | TResult
+  // so it needs to be awaited
+  return await syncedHandler(sampleEvent, sampleContext, () => {})
```

Comment:
Called it out in the comments, but calling it out again to say that this needs to be awaited just because it could be async.